src/docs/development/add-to-app/ios/add-flutter-screen.md

```diff
@@ -6,77 +6,291 @@ description: Learn how to add a single Flutter screen to your existing iOS app.
 
 This guide describes how to add a single Flutter screen to an existing iOS app.
 
-## Write code to use FlutterViewController from your host app
+## Start a FlutterEngine and FlutterViewController
 
-The proper place to do this is specific to your host app. Here is an example
-that makes sense for the blank screen of the host app generated by Xcode.
+Launching a Flutter screen from an existing iOS comprises of starting a
+[`FlutterEngine`]({{site.api}}/objcdoc/Classes/FlutterEngine.html)
+and a [`FlutterViewController`]({{site.api}}/objcdoc/Classes/FlutterViewController.html).
 
-First declare your app delegate to be a subclass of `FlutterAppDelegate`. Then define a FlutterEngine property, which helps you to register a plugin without a FlutterViewController instance.
+{{site.alert.secondary}}
+  The `FlutterEngine` serves as a host to the Dart VM and your Flutter runtime
+  and the `FlutterViewController` attaches to a `FlutterEngine` to pass UIKit
+  input events into Flutter and to display frames rendered by the `FlutterEngine`.
+{{site.alert.end}}
 
-In `AppDelegate.h`:
+The `FlutterEngine` may have the same lifespan as your `FlutterViewController`
+or outlive your `FlutterViewController`.
 
-```objective-c
+{{site.alert.tip}}
+It's generally recommended pre-warm a long-lived `FlutterEngine` for your
+application. This way,
+
+- Flutter screens will render their first frames faster.
+- Your Flutter and Dart state could outlive one `FlutterViewController`.
+- You and your plugins can interact with Flutter and your Dart logic before
+  showing UI.
+{{site.alert.end}}
+
+See [Loading sequence and performance](/docs/development/add-to-app/performance)
+for more analysis on the latency and memory trade-offs of pre-warming an engine.
+
+### Create a FlutterEngine
+
+The proper place to do this is specific to your host app. As an example, we will
+demonstrate creating a `FlutterEngine`, exposed as a property, on app startup in
+the app delegate.
+
+<ul class="nav nav-tabs" id="engine-language" role="tablist">
```

Comment:
@matthew-carroll see staging at https://add-to-app-2019.firebaseapp.com/docs/development/add-to-app/ios/add-flutter-screen. 

This seems like a reasonable way of toggling languages for instructions in a way consistent with the aesthetics of the rest of the website.